src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinFeature.kt

```diff
@@ -42,7 +42,23 @@ enum class KotlinFeature(private val enabledByDefault: Boolean) {
      * may contain null values after deserialization.
      * Enabling it protects against this but has significant performance impact.
      */
-    StrictNullChecks(enabledByDefault = false);
+    StrictNullChecks(enabledByDefault = false),
+
+    /**
+     * By enabling this feature, the property name on Kotlin will be used as the getter name.
+     *
+     * By default, the name based on the getter name on the JVM is used as the accessor name.
+     * This name may be different from the parameter/field name, in which case serialization results
+     * may be incorrect or annotations may malfunction.
+     * See [jackson-module-kotlin#630] for details.
+     *
+     * By enabling this feature, such malfunctions will not occur.
+     *
+     * On the other hand, enabling this option increases the amount of reflection processing,
+     * which may result in performance degradation for both serialization and deserialization.
+     * In addition, the adjustment of behavior using get:JvmName is disabled.
+     */
+    UseKotlinPropertyNameForGetter(enabledByDefault = false);
```

Comment:
@cowtowncoder 
I would like to get a review as I am not sure about the option name and description.

Reply:
One quick question: is "property" first-class concept in Kotlin? If not, maybe it'd be "field", but I assume it is.

But I guess I am not sure what "use property name for getter" actually means... I mean, getter in Jackson specifically means method use to access value of a logical property to serialize. It has name specified by class file, and does not change.
Name of logical property does change tho. So maybe naming here is bit confusing, at least to me.

Reply:
> is "property" first-class concept in Kotlin?

I think we can call it a first class concept in this regard.
On `Kotlin`, fields and accessors are defined for properties.
https://kotlinlang.org/docs/properties.html

> So maybe naming here is bit confusing, at least to me.

I don't know if this is a good name, but at least this is how I would describe the internal behavior directly.

If the emphasis is on naming on `Jackson`, would it be `UseKotrinPropertyNameForGetterImplicitName`?
However, it doesn't seem like a good name since we may not use the `findImplicitPropertyName` function forever.
Also, the name seems too long compared to the traditional `KotlinFeature`.

Reply:
I just did not understand what "for getter" means. But yes, if it changes "implicit [property] name" in context of serialization, I can see how "....ForGetterImplicitName" could work.

Does this not affect deserialization? Otherwise just something like "UseKotlinNameAsImplicitName" would make more sense to me.

Reply:
> I just did not understand what "for getter" means.

I wanted to express that it does not affect parameters, fields, or setters, but only getters.
Currently we are already using names on `Kotlin` for parameter names, but that is not affected by this option.
The same applies to fields and setters.

> Does this not affect deserialization?

If there are annotations for deserialization on the `getter`, it will affect the behavior, but basically it should only affect serialization.

Reply:
Right. I think referring to serialization makes more sense because getter really is physical thing (Method) used to access a property for purpose of (de)serialization. So you can't really change name of getter, but you change name of property for (de)serialization. Although, yes, obtained from getter method.

Still, ultimately this is your decision: I just offer my suggestions. :)

Reply:
Pushed for correction.
[aa7217f](https://github.com/FasterXML/jackson-module-kotlin/pull/686/commits/aa7217f4ba0ab09896f5806d6ce9458c072c319f)